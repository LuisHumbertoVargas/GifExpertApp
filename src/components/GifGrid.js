import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
// import React, { useEffect, useState } from 'react';
// import { getGifs } from '../helpers/getGifs';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs(category).then(setImages);
    // }, [category]);

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeInTopLeft'>
                {category}
            </h3>
            {loading && (
                <h3 className='animate__animated animate__flash'>Loading...</h3>
            )}
            <div className='card-grid'>
                {images.map((img) => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};

export default GifGrid;
