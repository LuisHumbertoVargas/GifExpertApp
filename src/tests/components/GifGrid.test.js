import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba unitaria en el componente <GifGrid />', () => {
    const category = 'category';
    test('Debe de hacer el snapShot del componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes "useFetchGifs"', () => {
        const gifs = [
            {
                id: 'ChanchitoFeliz',
                url: 'https://localhost/chanchitoFeliz/image.gif',
                title: 'Cualquier cosa',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
