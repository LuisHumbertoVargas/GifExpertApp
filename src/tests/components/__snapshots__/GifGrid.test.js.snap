// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba unitaria en el componente <GifGrid /> Debe de hacer el snapShot del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInTopLeft"
  >
    category
  </h3>
  <h3
    className="animate__animated animate__flash"
  >
    Loading...
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba unitaria en el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes "useFetchGifs" 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInTopLeft"
  >
    category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ChanchitoFeliz"
      key="ChanchitoFeliz"
      title="Cualquier cosa"
      url="https://localhost/chanchitoFeliz/image.gif"
    />
  </div>
</Fragment>
`;
